src/backend/gpopt/translate/CQueryMutators.cpp

```diff
@@ -798,17 +799,46 @@ CQueryMutators::RunExtractAggregatesMutator(Node *node,
 
 		GPOS_ASSERT(IsA(old_sublink->subselect, Query));
 
-		new_sublink->subselect = gpdb::MutateQueryOrExpressionTree(
-			old_sublink->subselect,
-			(MutatorWalkerFn) RunExtractAggregatesMutator, (void *) context,
-			0  // mutate into cte-lists
-		);
+		// One need to call the Query mutator for subselect and take into
+		// account that SubLink can be multi-level. Therefore, the
+		// context->m_current_query_level must be modified properly
+		// while diving into such nested SubLink.
+		new_sublink->subselect =
+			RunExtractAggregatesMutator(old_sublink->subselect, context);
 
 		context->m_current_query_level--;
 
 		return (Node *) new_sublink;
 	}
 
+	if (IsA(node, Query))
+	{
+		Query *query = gpdb::MutateQueryTree(
+			(Query *) node,
+			(MutatorWalkerFn) CQueryMutators::RunExtractAggregatesMutator,
+			context,
+			QTW_IGNORE_RT_SUBQUERIES  // flag -- do not mutate range table entries
```

Comment:
This flag lets to ignore RTE_SUBQUERY only, but the comment says that all RTEs are not mutated